apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts

```diff
@@ -67,14 +67,11 @@ domReady( () => {
 				dispatch( 'automattic/launch' ).openSidebar();
 			}
 
-			// @TODO: this is just temporary for testing via feature flag. Remove it once focused-launch is live
-			const isFocusedLaunchFlowFeatureFlag = window.calypsoifyGutenberg?.isFocusedLaunchFlow;
-
-			if ( launchFlow === FOCUSED_LAUNCH_FLOW || isFocusedLaunchFlowFeatureFlag ) {
+			if ( launchFlow === FOCUSED_LAUNCH_FLOW ) {
 				dispatch( 'automattic/launch' ).openFocusedLaunch();
 			}
 
-			if ( launchFlow === SITE_LAUNCH_FLOW && ! isFocusedLaunchFlowFeatureFlag ) {
+			if ( launchFlow === SITE_LAUNCH_FLOW ) {
```

Comment:
I would either:
- add a `return`  clause inside all these ifs blocks, or
- re-write the ifs as a `if (..) { … } else if (…) else { … }` chain, or
- use a `switch` statement instead for all the `launchFlow` checks

Basically, this would be to make it explicit that only one type of launch can happen at the same time.

Bonus question: what does the `dispatch( ‘automattic/launch’ ).enableAnchorFm()` call do? If that line is called, should we still run through the rest of the file, or can we _early `return`_?

Reply:
> add a `return` clause inside all these ifs blocks,

Possible with some wrangling... there is a part at the end which calls `savePost()`.
This happens in the background for gutenboardingLaunch and focusedLaunch, whereas, this needs to be happen first before siteLaunch.

> use a `switch` statement instead for all the launchFlow checks

Possible too, but I think I'm in favour of separate `if` blocks. Not have to add `break;` is nice.


> Bonus question: what does the dispatch( ‘automattic/launch’ ).enableAnchorFm() call do? If that line is called, should we still run through the rest of the file, or can we early return?

Anchor-flavoured sites skip some steps in the gutenboarding launch flow (and they are exclusively used the gutenboarding launch flow). `enableAnchorFm` toggles a flag in the data store that changes the step sequence, then the step-by-step launch opens.

Reply:
I see, thank you for the explanation

> there is a part at the end which calls savePost(). This happens in the background for gutenboardingLaunch and focusedLaunch, whereas, this needs to be happen first before siteLaunch.

I missed that part! In that case, we can probably leave it as is — I'm not sure that any refactor would improve the code clarity much

Reply:
I'm going to leave the `switch` version here for reference.
I'm not sure why there was a `setTimeout` on the `savePost()` for gutenboarding launch & focused launch.
It's either because we don't want user to see the post is saving or we don't want 2 things to happen at once.
This switch version reuses one `savePost()` for all without the `setTimeout`.

```js
const saveTask = dispatch( 'core/editor' ).savePost();

switch ( launchFlow ) {
	case GUTENBOARDING_LAUNCH_FLOW:
		dispatch( 'automattic/launch' ).openSidebar();
		break;

	case FOCUSED_LAUNCH_FLOW:
		dispatch( 'automattic/launch' ).openFocusedLaunch();
		break;

	case SITE_LAUNCH_FLOW:
		// Save post first before redirecting to launch url
		( async () => {
			await saveTask;
			window.top.location.href = launchUrl;
		} )();
		break;
}
```

Reply:
I think that the delay was there to:
- prioritise any API call that the launch modal would make
- avoid showing the "post saved" toast in the bottom left of the screen (by adding a delay, the toast would appear while the launch modal is already visible, and therefore the user would not see it)

This is one possible way of using a `switch` statement while keeping the `setTimeout` call

```typescript
const { savePost } = dispatch( 'core/editor' );
const delayedSavePost = () => setTimeout( () => savePost(), 1000 );

switch ( launchFlow ) {
	case GUTENBOARDING_LAUNCH_FLOW:
		// Save post first before redirecting to launch url
		dispatch( 'automattic/launch' ).openSidebar();
		delayedSavePost();
		break;
	case FOCUSED_LAUNCH_FLOW:
		// Save post first before redirecting to launch url
		dispatch( 'automattic/launch' ).openFocusedLaunch();
		delayedSavePost();
		break;
	case SITE_LAUNCH_FLOW:
		( async () => {
			await savePost();
			window.top.location.href = launchUrl;
		} )();
		break;
}
```

Reply:
Actually, I managed to track down the PR that introduced the 1s delay #44909

Reply:
Thanks for the `switch` version which seems like a good middle ground for now. :) Delaying save post does have its benefit regardless of the reason, and this block of code is readable, much readable than the existing one where it's executed at the bottom. I'm committing this in.

Reply:
I don't remember exactly why it's 1s delay. Maybe something related to how long it would take for the launch modal to open? Eg: with a slide-in animation

Reply:
Alright — let's keep the 1s delay in, for now